### PR TITLE
- improved folder deletion logic. previously only files were deleted not the folders.

### PR DESCRIPTION
**Pull request by adhudiya**
- if there is a empty folder then it was skipping from deletion.

### CODE REVIEW

File: src/main/java/com/celarli/commons/vfs/provider/google/GCSFileObject.java [hunk 1]
```diff
@@ -128,6 +128,12 @@ protected FileType doGetType() throws Exception {
                 return FileType.FOLDER;
             }
             else {
+
+                //blobs are not listed if url starts with /.
+                if (url.startsWith("/")) {
```
Comment (ashokmor) [inline src/main/java/com/celarli/commons/vfs/provider/google/GCSFileObject.java:133]: @adhudiya 

what happens if user defined root location? (like / )

Comment (adhudiya) [reply src/main/java/com/celarli/commons/vfs/provider/google/GCSFileObject.java:133]: in that situation if block will be executed not the else

Comment (ashokmor) [reply src/main/java/com/celarli/commons/vfs/provider/google/GCSFileObject.java:133]: why are we removing leading slash ?

Comment (adhudiya) [reply src/main/java/com/celarli/commons/vfs/provider/google/GCSFileObject.java:133]: internally it doesn't list blobs if leading slash is there, you will find same code on line number 107

**Pull request merged**